dashboard/app/models/script.rb

```diff
@@ -426,25 +426,30 @@ def self.get_without_cache(id_or_name, version_year: nil)
   # @param id_or_name [String|Integer] script id, script name, or script family name.
   # @param version_year [String] If specified, when looking for a script to redirect
   #   to within a script family, redirect to this version rather than the latest.
-  def self.get_from_cache(id_or_name, version_year: nil)
-    return get_without_cache(id_or_name, version_year: version_year) unless should_cache?
+  def self.get_from_cache(id_or_name, version_year: nil, user: nil, locale: nil)
```

Comment:
Apologies Maddie, but I'm afraid this approach has some fatal flaws:
1. the user and locale are being used to lookup the script, but then they are being cached using a `cache_key` which does not include the user and locale. this means that whoever looks up a script first (e.g. Maddie in spanish) will have their result cached and the next user (Dave in Hindi) will be sent to coursea 2017 (Maddie's default) instead of 2018 (Dave's expected default).
2. practically speaking, you can't work around this by adding the user and locale to the cache key, because that will blow up the size of the cache dramatically.

If I had to suggest a new approach to explore, it would be to keep user and locale outside of Script.get_with_cache, and then in the ScriptsController, fetch all scripts in this family (should be cheap since they are cached). then use the user and locale to decide whether to serve the current one or redirect to one of the others.

Sorry for leaving you this tripwire -- it seemed like a good idea to use the script cache to cache the redirect at the time, but now it does not seem practical any more. Please feel free to ping me if you'd like to discuss prior to taking another crack at this.

Reply:
don't apologize -- thank you for catching this! i'll rethink this approach and ping you if i have any questions 😄